### PR TITLE
refactor!: move transaction status and type to ape-ethereum

### PR DESCRIPTION
**Pull request by unparalleled-js**
BREAKING CHANGE: Moves `TransactionStatusEnum` and `TransactionType` from `ape.api.transactions` to `ape_ethereum.transactions`.

### What I did

Right now, the API assumes you use Ethereum-esque transaction status numbers and types.
This PR lessens those assumptions by counting on primitive values in the API and moves the Enum classes to the `ape-ethereum` core plugin.

One weird part is the `Web3Provider` mixin still needs to check the transaction type in an Ethereum context (`Web3Provider` is EVM by nature so this is okay I think).

### How I did it

* Create a `ape_ethereum.transactions` module that contains the transaction implementations, the receipt, and now the enums that were originally in `ape.api`

### How to verify it

You can still make transactions!
I am going to leave this as a draft to ensure it works in the Starknet land.

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: tests/functional/geth/test_providers.py [hunk 1]
```diff
@@ -1,74 +0,0 @@
...
-from web3.exceptions import ContractLogicError as Web3ContractLogicError
-
-from ape.exceptions import ContractLogicError, TransactionError
-from ape_geth import GethProvider
-
-_TEST_REVERT_REASON = "TEST REVERT REASON."
-
-
-class TestEthereumProvider:
-    def test_send_when_web3_error_raises_transaction_error(
```
Comment (fubuloubu) [inline tests/functional/geth/test_providers.py:13]: Why was this file deleted?

**Review by fubuloubu — APPROVED** [review]

File: tests/functional/geth/test_providers.py (hunk 1 above)
Comment (unparalleled-js) [reply tests/functional/geth/test_providers.py]: oops!

**Review by fubuloubu — APPROVED** [review]

**Review by NotPeopling2day — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]
